### PR TITLE
Tonitec

### PR DESCRIPTION
**Pull request by onedot-webagency**
# Pull Request Template

## Description

Added support for the SwagCustomProducts plugin by exporting the custom attributes set by it and enabling the attribute customization during checkout.

## Type of change

- [ ] Bug fix
- [X] New feature
- [ ] This change requires a documentation update

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [x] I updated the CHANGELOG.md

### CODE REVIEW

File: src/Backend/SgateShopgatePlugin/Components/User.php [hunk 1]
```diff
@@ -123,15 +123,6 @@ public function loginUser($request, $httpResponse)
...
         } else {
-            if (!empty($basket['content'])) {
-                $this->basket->clearBasket();
-                $this->basket->sRefreshBasket();
-
-                foreach ($basket['content'] as $basketItem) {
-                    $this->basket->sAddArticle($basketItem['ordernumber'], $basketItem['quantity']);
-                }
-            }
-
```
Comment (onedot-webagency) [inline src/Backend/SgateShopgatePlugin/Components/User.php:134]: This change was in response to a bug which appeared during checkout: The articles were added to the cart but the custom attributes were lost. This was found to be caused by the marked code clearing the cart and adding all articles again, the second time without the attribute customization. Once the cart is cleared, the customized attributes can no longer be restored. 

As there is no obvious reason for this code, it was decided to remove it to fix the aforementioned bug.

File: src/Backend/SgateShopgatePlugin/Helpers/WebCheckout.php [hunk 2]
```diff
@@ -85,19 +85,6 @@ public function loginAppUser($token, $request)
...
-            if (!empty($basket['content'])) {
-                foreach ($basket['content'] as $basketItem) {
-                    Shopware()->Modules()->Basket()->sAddArticle($basketItem['ordernumber'], $basketItem['quantity']);
-                }
-
-                if (!empty($voucher)) {
-                    Shopware()->Modules()->Basket()->sAddVoucher($voucher['code']);
-                }
-            }
-
```
Comment (onedot-webagency) [inline src/Backend/SgateShopgatePlugin/Helpers/WebCheckout.php:100]: This change was in response to a bug which appeared during checkout: The articles were added to the cart but the custom attributes were lost. This was found to be caused by the marked code clearing the cart and adding all articles again, the second time without the attribute customization. Once the cart is cleared, the customized attributes can no longer be restored. 

As there is no obvious reason for this code, it was decided to remove it to fix the aforementioned bug.

File: src/Backend/SgateShopgatePlugin/Models/Export/Product.php [hunk 3]
```diff
@@ -656,6 +656,120 @@ public function getParentCategoryIds(\Shopware\Models\Category\Category $cat)
...
+                `price`.`customer_group_name` AS price_customer_group_name,
+                `price`.`customer_group_id` AS price_customer_group_id,
+                `price`.`is_percentage_surcharge` AS price_is_percentage_surcharge,
+                `price`.`percentage` AS price_percentage
+
+            FROM
+                s_plugin_custom_products_template `template`
+                LEFT JOIN s_plugin_custom_products_template_product_relation `r` ON `r`.`template_id` = `template`.id
+                LEFT JOIN s_plugin_custom_products_option `option` ON `option`.`template_id` = `template`.id
+                LEFT JOIN s_plugin_custom_products_price `price` ON `price`.`option_id` = `option`.`id`
```
Comment (onedot-webagency) [inline src/Backend/SgateShopgatePlugin/Models/Export/Product.php:725]: These JOINs only capture simple input fields from the plugin (including file-uploads). Dropdown-based customized attributes are not supported.

**Comment by Rekhyt** [comment]
These are merchant specific changes that won't be merged into mainline. I was told that the PR was opened for diff & comments only. Going to close it, feel free to open another and reference this one if required.

**Pull request closed without merging**